packages/back-end/src/models/VisualChangesetModel.ts

```diff
@@ -19,11 +19,29 @@ import {
   updateExperiment,
 } from "./ExperimentModel";
 
+const visualChangesetURLPatternSchema = new mongoose.Schema<VisualChangesetURLPattern>(
+  {
+    include: Boolean,
+    type: {
+      type: String,
+      enum: ["simple", "regex"],
+      required: true,
+    },
+    pattern: {
+      type: String,
+      required: true,
+    },
+  },
+  {
+    _id: false,
+  }
+);
```

Comment:
It would be great to check that these changes do not introduce a regression. The existing `urlPatterns` definition used `_id: false` which gives a type error (invalid property).  There's also a property `id: boolean`, not sure if that's relevant.

Reply:
@bttf would you be able to test that this change does not introduce any regressions to the visual changesets?

Reply:
Tested; no regressions

(The reason this occurred is because we couldn't use shorthand syntax for the schema definition of urlPatterns since the it has a `type` key, which is reserved by mongoose when defining fields. The way you've done it by separating it out into its own schema avoids this.)

Reply:
Thanks for testing, I appreciate it!